File: HRMS/HRMS.Web/Modules/Operations/Holiday/HolidayColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.Holiday")]
[BasedOnRow(typeof(HolidayRow), CheckNames = true)]
public class HolidayColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int HolidayId { get; set; }
    [EditLink]
    public string HolidayName { get; set; }
    public DateTime HolidayDate { get; set; }
    public HolidayType HolidayType { get; set; }
    public string Branch { get; set; }
    public bool IsOptionalHoliday { get; set; }
    public int Year { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/HolidayColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { HolidayRow } from "./HolidayRow";
import { HolidayType } from "./HolidayType";

export interface HolidayColumns {
    HolidayId: Column<HolidayRow>;
    HolidayName: Column<HolidayRow>;
    HolidayDate: Column<HolidayRow>;
    HolidayType: Column<HolidayRow>;
    Branch: Column<HolidayRow>;
    IsOptionalHoliday: Column<HolidayRow>;
    Year: Column<HolidayRow>;
}

export class HolidayColumns extends ColumnsBase<HolidayRow> {
    static readonly columnsKey = 'Operations.Holiday';
    static readonly Fields = fieldsProxy<HolidayColumns>();
}

[HolidayType]; // referenced types